packages/aquedux-client/src/middlewares/client.js

```diff
@@ -0,0 +1,18 @@
+// @flow
+
+import actionTypes from '../constants/actionTypes'
+import * as eventHub from '../utils/eventHub'
+
+export default (_store: any) => (next: Function) => (action: Object) => {
+  const { type } = action
+
+  if (type === actionTypes.AQUEDUX_CLIENT_START) {
+    eventHub.raise(eventHub.AQUEDUX_CLIENT_START)
+  } else if (type === actionTypes.AQUEDUX_CLIENT_STOP) {
+    eventHub.raise(eventHub.AQUEDUX_CLIENT_STOP)
+  } else if (type === actionTypes.AQUEDUX_CLIENT_RESTART) {
+    eventHub.raise(eventHub.AQUEDUX_CLIENT_RESTART)
+  }
```

Comment:
If using the same constant is not possible, I would suggest using a map:
```
const eventMap = {
  actionTypes.AQUEDUX_CLIENT_START: eventHub.AQUEDUX_CLIENT_START,
  actionTypes.AQUEDUX_CLIENT_STOP: eventHub.AQUEDUX_CLIENT_STOP
}

export default (_store: any) => (next: Function) => (action: Object) => {
  const { type } = action
  const event = eventMap[type]
  event && eventHub.raise(event)
  return next(action)
}

Reply:
Even if I like the map solution, we should keep the if-else-if statements for code simplicity IMO. If we switch the eventHub with something else in the future (observables ? 🏇 ) we'll take care of that piece of code.

And I think it is better to have an explicit set of eventTypes because we do not expect to `raise` all the actionTypes, only those 3 (well maybe 2 soon if the restart is removed).

Reply:
No pb to keep this, especially the restart action will be removed

Reply:
👍